src/js/containers/account/visualizations/AccountTimeVisualizationContainer.jsx

```diff
@@ -103,52 +148,92 @@ export class AccountTimeVisualizationSectionContainer extends React.Component {
     }
 
     parseBalances(data) {
-        const years = [];
-
-        const balances = {};
-        data.forEach((item, i) => {
-            const type = this.balanceRequests[i].type;
-            const values = {};
-
-            item.data.results.forEach((group) => {
-                const date = moment(group.item, 'YYYY-MM-DD');
-                const fy = FiscalYearHelper.convertDateToFY(date);
-                if (_.indexOf(years, fy) === -1) {
-                    years.push(fy);
-                }
-                values[fy] = group.aggregate;
-            });
-
-            balances[type] = values;
-        });
-
-
         const groups = [];
         const xSeries = [];
         const ySeries = [];
+        const allY = [];
+        if (this.state.hasFilteredObligated) {
+            // In progress
+        }
+        else {
+            // does not have filtered obligated
+            if (this.state.visualizationPeriod === 'quarter') {
+                const quarters = [];
+                const yData = [];
+                data.forEach((balance, index) => {
+                    balance.data.results.forEach((group, index2) => {
+                        if (index == 0) {
+                            quarters.push(group.item + " Q" + group.submission__reporting_fiscal_quarter);
+                            yData.push({});
+                            yData[index2].outlay = parseFloat(group.aggregate);
+                        }
+                        // is there a more efficient way to do this?
```

Comment:
Yes there is. `this.balanceRequests` is an array of objects. For each API call that is made:
```
{
    cancel: // function to cancel the API call
    promise: // the JS promise that contains the API call
    type: "outlay" // a string indicating the data type that lines up with the property you are setting
}
```

So you can simply do:
```
const type = this.balanceRequests[index2].type;
yData[index2][type] = parseFloat(group.aggregate);
```